src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -3022,6 +2971,87 @@ export function updateSavedUserPreferenceChartIds(
     return chartSettings;
 }
 
+export async function getAllClinicalDataByStudyViewFilter(
+    studyViewFilter: StudyViewFilter
+): Promise<{ [sampleId: string]: { [attributeId: string]: string } }> {
+    const localClinicalDataCollection: ClinicalDataCollection = {
+        sampleClinicalData: [],
+        patientClinicalData: [],
+    };
+    let remoteClinicalDataCollection: ClinicalDataCollection = {
+        sampleClinicalData: [],
+        patientClinicalData: [],
+    };
+
+    const maxPageSize = 500000;
+    let pageNumber = 0;
+
+    do {
+        const remoteClinicalDataCollection = await internalClient.fetchClinicalDataClinicalTableUsingPOST(
+            {
+                studyViewFilter,
+                pageSize: maxPageSize,
+                pageNumber: pageNumber,
+                searchTerm: undefined,
+                sortBy: undefined,
+                direction: 'ASC',
+            }
+        );
+        localClinicalDataCollection.sampleClinicalData = localClinicalDataCollection.sampleClinicalData.concat(
+            remoteClinicalDataCollection.sampleClinicalData
+        );
+        localClinicalDataCollection.patientClinicalData = localClinicalDataCollection.patientClinicalData.concat(
+            remoteClinicalDataCollection.patientClinicalData
+        );
+        pageNumber++;
+    } while (remoteClinicalDataCollection.sampleClinicalData.length > 0);
+
+    return mergeClinicalDataCollection(localClinicalDataCollection);
+}
+
+export function mergeClinicalDataCollection(
+    clinicalDataCollection: ClinicalDataCollection
+): { [sampleId: string]: { [attributeId: string]: string } } {
+    const patientKeyedSampleData = _.groupBy(
+        clinicalDataCollection.sampleClinicalData,
+        d => d.uniquePatientKey
+    );
+    let patientKeyedSampleKeyedData = _.mapValues(
+        patientKeyedSampleData,
+        sampleData => _.groupBy(sampleData, d => d.uniqueSampleKey)
+    );
+    const patientKeyedPatientData = _.groupBy(
+        clinicalDataCollection.patientClinicalData,
+        d => d.uniquePatientKey
+    );
+    // Add patient level clinical data to sample clinical data.
+    patientKeyedSampleKeyedData = _.mapValues(
+        patientKeyedSampleKeyedData,
+        (sampleKeyedData, patientId) =>
+            _.mapValues(sampleKeyedData, (attrs, sampleId) =>
+                attrs.concat(patientKeyedPatientData[patientId] || [])
+            )
+    );
+    // Remove patient id levels (only keep sample id keys).
+    const sampleKeyedData = _.assign(
+        {},
+        ..._.values(patientKeyedSampleKeyedData)
+    );
+    // Put all clinical attributes in one object.
+    const sampleCollapsedAttributes = _.mapValues(
+        sampleKeyedData,
+        clinicalData => {
+            const data = _.map(clinicalData, (datum: ClinicalData) => {
+                const obj = {} as { [attrId: string]: string };
```

Comment:
const data: Record<string, string> = {} is stronger than using "as"

Reply:
Ah I did not know. Updated.